### PR TITLE
refactor: 💡 Add an id to the service to handle name changes

### PR DESCRIPTION
**Pull request by MaethorNaur**
Add an id to the Service, and front events to make services unique

BREAKING CHANGE: 🧨 Service and front Events now have an id field

closes: #20

### CODE REVIEW

File: core/src/main/scala/restui/models/Event.scala [hunk 1]
```diff
@@ -2,26 +2,27 @@ package restui.models
...
+          "id"       -> Json.fromString(id),
+          "name"     -> Json.fromString(name),
           "metadata" -> Json.obj(metadata.view.mapValues(Json.fromString).toSeq: _*)
         )
-      case ServiceDown(serviceName) =>
+      case ServiceDown(id) =>
         Json.obj(
           "event" -> Json.fromString("serviceDown"),
-          "name"  -> Json.fromString(serviceName)
+          "name"  -> Json.fromString(id)
```
Comment (andrewregas) [inline core/src/main/scala/restui/models/Event.scala:25]: You want this to be a name in Json ?

File: providers/docker/src/main/scala/restui/providers/docker/DockerClient.scala [hunk 2]
```diff
@@ -68,7 +68,7 @@ class DockerClient(private val client: JDockerClient, private val settings: Sett
           findEndpoint(labels, networks).map {
             case (serviceName, address) =>
               if (event.getStatus == StartFilter)
-                ServiceEvent.ServiceUp(Service(serviceName, OpenApiFile(ContentType.fromString(address), address)))
+                ServiceEvent.ServiceUp(Service(serviceName, serviceName, OpenApiFile(ContentType.fromString(address), address)))
```
Comment (andrewregas) [inline providers/docker/src/main/scala/restui/providers/docker/DockerClient.scala:71]: you pass serviceName twice for id and servicename is that what you want?

File: core/src/main/scala/restui/models/Event.scala (hunk 1 above)
Comment (MaethorNaur) [reply core/src/main/scala/restui/models/Event.scala]: Thanks

File: providers/docker/src/main/scala/restui/providers/docker/DockerClient.scala (hunk 2 above)
Comment (MaethorNaur) [reply providers/docker/src/main/scala/restui/providers/docker/DockerClient.scala]: I moved to the container id, I did the same for the Kubernetes provider

**Pull request merged**